hw_05/Program.cs

```diff
@@ -0,0 +1,147 @@
+            int[] userArr = FillArrayByUser(new int[arrSize], 0, arrSize);
+
+            int[] resultArr = new int[arrSize];
+            for (int i = 0; i < arrSize; i++)
+            {
+                resultArr[i] = userArr[i] + randomArr[i];
+            }
+
+            Console.WriteLine("Random array values:");
+            PrintArray(randomArr);
+            Console.WriteLine();
+
+            Console.WriteLine("User array values:");
+            PrintArray(userArr);
+            Console.WriteLine();
+
+            Console.WriteLine("Result array values:");
+            PrintArray(resultArr);
+            Console.WriteLine();
+        }
+
+        public static void Task2()
+        {
+            int arrSize = 10;
+            int arrFillSize = 9;
+
+            Console.WriteLine($"Enter {arrFillSize} integer to fill array");
+            int[] userArr = FillArrayByUser(new int[arrSize], 0, arrFillSize);
+
+            Console.WriteLine("Initial array:");
+            PrintArray(userArr);
+            Console.WriteLine();
+
+            Console.WriteLine($"Enter insert index from 0 to {arrSize - 1}:");
+            int insertIndex = Convert.ToInt32(Console.ReadLine());
+
+            Console.WriteLine("Enter integer to insert");
+            int insertValue = Convert.ToInt32(Console.ReadLine());
+
+            Insert(userArr, insertIndex, insertValue);
+
+            Console.WriteLine("Result array:");
+            PrintArray(userArr);
+        }
+
+        public static void Task3()
+        {
+            int arrSize = 11;
+            int[] randomArr = FillArrayRandom(new int[11], 0, arrSize);
+
+            Console.WriteLine("Initial array:");
+            PrintArray(randomArr);
+            Console.WriteLine();
+
+            Reverse(randomArr);
+
+            Console.WriteLine("Result array:");
+            PrintArray(randomArr);
+
+        }
+
+        public static void PrintArray(int[] arr)
+        {
+            for (int i = 0; i < arr.Length; i++)
+            {
+                Console.Write($"{arr[i]} ");
+            }
+        }
+
+        public static int[] Insert(int[] arr, int index, int value)
+        {
+            for (int i = arr.Length - 1; i >= index; i--)
+            {
+                arr[i] = arr[i - 1];
+            }
+            arr[index] = value;
+
+            return arr;
+        }
+
+        public static int[] FillArrayByUser(int[] arr, int from, int to)
+        {
+            for (int i = from; i < to; i++)
+            {
+                arr[i] = Convert.ToInt32(Console.ReadLine());
+            }
+
+            return arr;
+        }
+
+        public static int[] FillArrayRandom(int[] arr, int from, int to)
+        {
+            Random randomGenerator = new Random();
+            for (int i = from; i < to; i++)
+            {
+                arr[i] = randomGenerator.Next(0, 100);
+            }
+
+            return arr;
+        }
+
+        public static T[] Reverse<T>(T[] arr)
```

Comment:
I see you have been using generic Type there. That is great! Have you discovered this approach yourself?

Reply:
I use generic  in TypeScript, it's very similar as i found out

Reply:
sure, this feature was derived from C#. 
all right!